my-profile-webapp/src/main/webapp/my-app/lec/partials/emergency-phone.html

```diff
@@ -21,7 +21,7 @@ <h2 class="md-headline">My phone number</h2>
         <br>
         <md-input-container class="md-block no-margin-bottom" flex-gt-sm>
           <label>Phone number*</label>
-          <input type="tel" required aria-required="true" ng-maxlength="30" ng-model="emergencyPhoneNumbers.emergencyPhoneNumbers[0].value">
+          <input type="tel" required aria-required="true" ng-pattern=/^(.*[a-zA-Z0-9\s!-.]){1,30}$/ ng-maxlength="30" ng-model="emergencyPhoneNumbers.emergencyPhoneNumbers[0].value">
```

Comment:
may not want to start with that .* ? I think that might greedily accept everything